http.go

```diff
@@ -26,31 +59,294 @@ func (*HTTP) CaddyModule() caddy.ModuleInfo {
+	if t.Metadata != "" {
+		t.opts = append(t.opts, config.WithMetadata(t.Metadata))
+	}
+
+	if t.AllowCIDR != nil {
+		t.opts = append(t.opts, config.WithAllowCIDRString(t.AllowCIDR...))
+	}
+
+	if t.DenyCIDR != nil {
+		t.opts = append(t.opts, config.WithDenyCIDRString(t.DenyCIDR...))
+	}
+
+	if t.CircuitBreaker != nil {
+		t.opts = append(t.opts, config.WithCircuitBreaker(*t.CircuitBreaker))
+	}
+
+	if t.EnableCompression {
+		t.opts = append(t.opts, config.WithCompression())
+	}
+
+	if t.Scheme != "" {
+		if t.Scheme == "http" {
+			t.opts = append(t.opts, config.WithScheme(config.SchemeHTTP))
+		} else if t.Scheme == "https" {
+			t.opts = append(t.opts, config.WithScheme(config.SchemeHTTPS))
+		}
+	}
+
+	if t.EnableWebsocketTCPConversion {
+		t.opts = append(t.opts, config.WithWebsocketTCPConversion())
+	}
+
+	for username, password := range t.BasicAuth {
+		t.opts = append(t.opts, config.WithBasicAuth(username, password))
+	}
+
+	return nil
+}
+
+func (t *HTTP) doReplace() error {
+	repl := caddy.NewReplacer()
+	replaceableFields := []*string{
+		&t.Metadata,
+		&t.Domain,
+		&t.Scheme,
+	}
+
+	for _, field := range replaceableFields {
+		actual := repl.ReplaceKnown(*field, "")
+
+		*field = actual
+	}
+
+	replacedAllowCIDR := make([]string, len(t.AllowCIDR))
+
+	for index, cidr := range t.AllowCIDR {
+		actual := repl.ReplaceKnown(cidr, "")
+
+		replacedAllowCIDR[index] = actual
+	}
+
+	t.AllowCIDR = replacedAllowCIDR
+
+	replacedDenyCIDR := make([]string, len(t.DenyCIDR))
+
+	for index, cidr := range t.DenyCIDR {
+		actual := repl.ReplaceKnown(cidr, "")
+
+		replacedDenyCIDR[index] = actual
+	}
+
+	t.DenyCIDR = replacedDenyCIDR
+
+	replacedBasicAuth := make(map[string]string, len(t.BasicAuth))
+
+	for username, password := range t.BasicAuth {
+		actualUsername := repl.ReplaceKnown(username, "")
+
+		actualPassword := repl.ReplaceKnown(password, "")
+
+		replacedBasicAuth[actualUsername] = actualPassword
+	}
+
+	t.BasicAuth = replacedBasicAuth
+
 	return nil
 }
 
 // convert to ngrok's Tunnel type
-func (h *HTTP) NgrokTunnel() config.Tunnel {
-	return config.HTTPEndpoint(h.opts...)
+func (t *HTTP) NgrokTunnel() config.Tunnel {
+	return config.HTTPEndpoint(t.opts...)
 }
 
 func (t *HTTP) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 	for d.Next() {
 		if d.NextArg() {
 			return d.ArgErr()
 		}
-		for d.NextBlock(0) {
-			switch d.Val() {
+
+		for nesting := d.Nesting(); d.NextBlock(nesting); {
+			subdirective := d.Val()
+			switch subdirective {
+			case "domain":
+				if !d.AllArgs(&t.Domain) {
+					return d.ArgErr()
+				}
+			case "metadata":
+				if !d.AllArgs(&t.Metadata) {
+					return d.ArgErr()
+				}
+			case "allow":
+				if err := t.unmarshalAllowCidr(d); err != nil {
+					return err
+				}
+			case "deny":
+				if err := t.unmarshalDenyCidr(d); err != nil {
+					return err
+				}
+			case "circuit_breaker":
+				if err := t.unmarshalCircuitBreaker(d); err != nil {
+					return err
+				}
+			case "enable_compression":
```

Comment:
The common pattern in Caddyfile for boolean toggles is to do something like `compression off`. See (Search `off`):

https://caddyserver.com/docs/caddyfile/options

https://caddyserver.com/docs/caddyfile/directives/reverse_proxy#the-http-transport

https://caddyserver.com/docs/caddyfile/directives/php_fastcgi#index